### ISSUE TITLE
Add support for specifying Altair data types in plotting functions

### ISSUE THREAD

**Issue by clausmith** [onecodex/onecodex#277]
For example, if `treatment_group` is a metadata group with values 1 - 6, we might want that to actually be encoded as a nominal variable for Altair. Typically, you'd just specify this as `treatment_group:N` when passing it to the `encode()` method. 

With our plotting helpers, we assume that the strings passed to args like `color` are metadata field names. We should add support for manually overriding the encoding type, so you could do something like this:

```
samples.plot_pca(
     color="treatment_group:N"
)
```

**Comment by polyatail** [comment]
Would this be solved if treatment_group were a string instead of an integer in the dataframe?

**Comment by clausmith** [comment]
Yes I believe so, but do we have a way of specifying whether a given metadata variable is a string/int in mainline? The issue arises from things being automatically coerced.

**Comment by boydgreenfield** [comment]
@clausmith Pandas does this automatically, but we could probably expose a `kwarg` to prevent this coercion for specific columns. Thoughts @polyatail?
